### PR TITLE
👾 Add proof of concept leak test 👾 

### PR DESCRIPTION
**Pull request by samccone**
I know this has been a longish time coming but here it is.
Ideally this will run in the same way that we run integration tests, based on the git diff.

This perf test is inline with all of my perf testing in that I have essentially codified the process here. 

https://github.com/samccone/leaky provides us with the instrumentation layer so we do not have to worry about it and get to reuse all of our selenium work in our integration suite :dancer: .

I currently have this hard coded to test against react, just to get some feedback on what everyone is  thinking.

------------

:fountain:  **React**:

```
initial snapshot: {"documents":3,"jsEventListeners":12,"jsHeapSizeUsed":9127312,"nodes":202}
end snapshot:     {"documents":3,"jsEventListeners":12,"jsHeapSizeUsed":10889192,"nodes":332}
```

node leak ^ as expected via https://github.com/tastejs/todomvc/issues/1332#issuecomment-115057932

:fountain: **Vue**

```
initial snapshot: {"documents":4,"jsEventListeners":7,"jsHeapSizeUsed":4696168,"nodes":293}
end snapshot:     {"documents":4,"jsEventListeners":7,"jsHeapSizeUsed":5681776,"nodes":383}
```

node leak ^ as expected via https://github.com/tastejs/todomvc/issues/1354

:fountain: **Angularjs**

```
initial snapshot: {"documents":3,"jsEventListeners":8,"jsHeapSizeUsed":6224976,"nodes":594}
end snapshot:     {"documents":3,"jsEventListeners":8,"jsHeapSizeUsed":8055816,"nodes":684}
```

node leak ^ as expected via https://github.com/tastejs/todomvc/issues/1335

:fountain: **Knockoutjs**

```
initial snapshot: {"documents":3,"jsEventListeners":7,"jsHeapSizeUsed":4346992,"nodes":273}
end snapshot:     {"documents":3,"jsEventListeners":37,"jsHeapSizeUsed":5414728,"nodes":363}
```

node and listener leak ^ as expected via https://github.com/tastejs/todomvc/issues/1334

:fountain: **Ember**

```
initial snapshot: {"documents":3,"jsEventListeners":5,"jsHeapSizeUsed":15816152,"nodes":267}
end snapshot:     {"documents":3,"jsEventListeners":5,"jsHeapSizeUsed":17672328,"nodes":517}
```

node leak ^ as expected via https://github.com/tastejs/todomvc/issues/1333

----------

*^ is by no means a comprehensive run of all of the implementations just the ones I have run*

### CODE REVIEW

**Comment by jasonLaster** [comment]
Woot woot. Seriously cool stuff

File: tests/leaks.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+var Page = require('./page');
+
+var driver = leaky.start({
+	chromeOptions: 'no-sandbox'
+});
+
+var page = new Page(driver);
+var initialSize = [];
+
+driver.get('http://localhost:8000/examples/react');
```
Comment (jasonLaster) [inline tests/leaks.js]: Hmm just react, yo?

Comment (arthurvr) [inline tests/leaks.js]: > I currently have this hard coded to test against react, just to get some feedback on what everyone is thinking.

**Comment by arthurvr** [comment]
This is really neat work :star: :100:

**Comment by addyosmani** [comment]
This is incredibly impressive work.

**Comment by arthurvr** [comment]
Really cool work! I'm going to close this for now as this was more of a proof of concept. We can come back to this when we've got a final version :)

**Pull request closed without merging**